### ISSUE TITLE
Adobe DC Reader pkcs11 support

### ISSUE THREAD

**Issue by 7lym** [jariq/pkcs11-mock#1]
Hello!
When i try to add your pkcs11 dll to adobe, it executes, in this order(as seen from a disassembler), calls to:
C_GetFunctionList, C_Initialize, C_GetSlotList, C_GetSlotList, C_GetSlotInfo, C_GetTokenInfo, C_GetSlotInfo, C_GetTokenInfo, C_GetTokenInfo, C_OpenSession, C_GetSlotInfo, C_GetTokenInfo and finally C_GetSessionInfo. After calling C_GetSessionInfo, it crashes with EXCEPTION_ACCESS_VIOLATION(when setting memory things in msvcr120.dll). Could you please check out the dll you provided?(i took the x86 one from your bin folder, as adobe is x86).Thanks!

**Comment by 7lym** [comment]
Also wanted to add that all functions return 0(CRK_OK) and in C_Initialize, the argument is nullptr, so I don\`t think it\`s on Adobe`s side.

**Comment by jariq** [comment]
I believe you might not have a correct understanding what pkcs11-mock is. It is by design that it returns "weird responses" because it is NOT a real cryptographic module but just a dummy mock object designed specifically for unit testing of Pkcs11Interop library ([read here for more info][1]). 

Maybe if you explain what you are trying to achieve I can point you to more suitable project (e.g. [SoftHSM][2]).

[1]: https://github.com/Pkcs11Interop/pkcs11-mock#overview
[2]: https://github.com/disig/SoftHSM2-for-Windows

**Comment by 7lym** [comment]
I\`m trying to build my own pkcs11 dll, and process the requests on a remote server(the server "is" the token). As I wanted to make sure the only thing I need in order for Adobe(or firefox etc) to properly use my dll are those ANSI C defined functions, I tested with your dll. I read the code, I knew it is a dummy dll, but I thought that on an information-gathering step(initialize, opening a session and feeding custom device info), your dll will do(of course I did not expect to sign anything or such:D). I basically hoped adobe would list the pkcs11 library on it\`s attached modules tab.
As I inspect the project you suggested, I guess it's close to what I intend to do. 
Thank you for your fast response! Will close the issue ticket now.
